bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/async/BulkMutationGCJClient.java

```diff
@@ -16,35 +16,36 @@
 package com.google.cloud.bigtable.grpc.async;
 
 import com.google.api.core.ApiFuture;
+import com.google.api.gax.batching.Batcher;
 import com.google.cloud.bigtable.config.Logger;
 import com.google.cloud.bigtable.core.IBulkMutation;
-import com.google.cloud.bigtable.data.v2.models.BulkMutationBatcher;
-import com.google.cloud.bigtable.data.v2.models.BulkMutationBatcher.BulkMutationFailure;
-import com.google.cloud.bigtable.data.v2.models.RowMutation;
+import com.google.cloud.bigtable.data.v2.models.RowMutationEntry;
 import com.google.cloud.bigtable.util.ApiFutureUtil;
 import com.google.common.base.Preconditions;
 
 /**
  * This class is meant to replicate existing {@link BulkMutation} while translating calls to
- * Google-Cloud-Java's {@link BulkMutationBatcher} api.
+ * Google-Cloud-Java's {@link Batcher} api.
  */
 public class BulkMutationGCJClient implements IBulkMutation {
 
   private static Logger LOG = new Logger(BulkMutation.class);
 
-  private final BulkMutationBatcher bulkMutateBatcher;
+  private final Batcher<RowMutationEntry, Void> bulkMutateBatcher;
   private final OperationAccountant operationAccountant;
 
-  public BulkMutationGCJClient(BulkMutationBatcher bulkMutateBatcher) {
+  public BulkMutationGCJClient(Batcher<RowMutationEntry, Void> bulkMutateBatcher) {
     this.bulkMutateBatcher = bulkMutateBatcher;
     this.operationAccountant = new OperationAccountant();
   }
 
   /** {@inheritDoc} */
   @Override
-  public synchronized ApiFuture<Void> add(RowMutation rowMutation) {
+  public synchronized ApiFuture<Void> add(RowMutationEntry rowMutation) {
     Preconditions.checkNotNull(rowMutation, "mutation details cannot be null");
     final ApiFuture<Void> response = bulkMutateBatcher.add(rowMutation);
+    // TODO(rahulkql): Still using operationAccount to give user isFlush() functionality. which
+    //  is not present in Batcher.
     operationAccountant.registerOperation(ApiFutureUtil.adapt(response));
```

Comment:
Please avoid using operationAccountant. It seems like this functionality is only used for the jvm shutdown hook to make sure there are no outstanding batches. We have this functionality in gcj already. Would it be possible move the shutdown hook to BulkMutationWrapper?

Reply:
By shutdown hook did you meant [this](https://github.com/googleapis/google-cloud-java/blob/master/google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/models/BulkMutationBatcher.java#L72-L84)? 

Also, I believe we don't need to wait till all the elements are finished as `Batcher#close` is going to flush it out. Please let me know If I am missing something here.

To provide `BulkMutationGCJClient#isFlush()` functionality I have used an AtomicLong#compare(0, 0), which I think should work. Please let me know if there could be a better approach?

Reply:
If I'm not mistaken, the only place that needs isFlushed() is:

https://github.com/googleapis/java-bigtable-hbase/blob/c4e970d347746f11aff082d3e87787b7e74eb59f/bigtable-client-core-parent/bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableConnection.java#L59-L77

And since we already check for unclosed batchers in gcj, I'm proposing removing the operation accountant from the gcj wrapper.

Please let me know if I'm missing something

Reply:
Yes, this was that is the only place where it has been used. I did remove it from `BulkMutationGCJClient` and replaced it with AtomicLong check. Please let me know if that looks good to you.